sympy/plotting/plot.py

```diff
@@ -2135,7 +2135,22 @@ def check_arguments(args, expr_len, nb_of_free_symbols):
             i = len(args) + 1
 
         exprs = Tuple(*args[:i])
-        free_symbols = list(set().union(*[e.free_symbols for e in exprs]))
+        free_symbols = list(set().union(*[e.free_symbols for e in exprs])) # sum(args[:expr_len]).free_symbols
+
+        #Warn: Dimension
+        variables = set(free_symbols) # get all variables
+        specify_symbols = set([s[0] for s in args[expr_len:]]) # get symbols from all ranges
+        all_symbols = variables | specify_symbols
+        if len(all_symbols) > len(variables):
+            msj = "Dimension Error: there be a variable ranged that ins´t part of any ecuation!"
+            raise Exception(msj)
```

Comment:
`Exception` shouldn't be used for exceptions. Rather use a more specific exception type (usually TypeError or ValueError, depending if the issue is with the input types or their values).